Changes after review

DamsYaYa opened the pull request:
Добавлены тесты для второй кнопки заказа
Тест проверяющий выпадающий список разбит на три самостоятельных

DamsYaYa committed:
Changes after review

pavel-ulan commented on src/test/java/DropDownList.java:
```diff
@@ -12,24 +12,27 @@
 
+        importantQuestions.checkImportantQuestionsTextSize();
+        importantQuestions.checkImportantQuestionsTextAvailability();
         importantQuestions.checkImportantQuestionsText();
```
**надо исправить**
нет, ты тут не разбила тест на самостоятельные. Ты просто обернула их в методы. А тест по сути не поменялся и все так же проверяет сначала к-во вопросов, потом доступность и наконец текст.
Давай сделаем проще - убери все проверки кроме проверки соответствия вопроса ответу.

pavel-ulan replied:
так же тест следовало сделать параметризованным, если ты хочешь реализовать все проверки через один тест. решение с циклом тут неприменимо (впрочем как и в любом другом тесте с несколькими параметрами допустима только параметризация).
если ты не можешь/не хочешь делать его параметризованным, значит разделяй на 8 отдельных тестов для проверки каждого вопроса отдельно.
Текущее решение - неправильное.

pavel-ulan commented on src/test/java/NewOrderTest.java:
```diff
@@ -59,4 +59,48 @@ public void testFirstWayOrderScooterInFirefox(){
         driver.quit();
     }
 
```
**надо исправить**
Где аннотация?

pavel-ulan commented on src/test/java/NewOrderTest.java:
```diff
@@ -59,4 +59,48 @@ public void testFirstWayOrderScooterInFirefox(){
+        WebDriverManager.chromedriver().setup();
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--no-sandbox", "--headless", "--disable-dev-shm-usage");
+        WebDriver driver = new ChromeDriver(options);
```
**надо исправить**
А почему в class DropDownList ты вынесла инициализацию драйвера в прекондишен, а тут нет?
Любые исправления должны проверяться по всей работе на аналогичные. Нашла проблему, исправила, проверила, что такихже больше в работе нет и только тогда заливаешь.

pavel-ulan replied:
еще момент. У тебя не должно быть 2 наборов тестов для разных браузеров. Браузер, это настройка. У тебя должна быть возможность использовать один браузер или второй для одних и тех же тестов.
это надо исправить.

pavel-ulan reviewed (changes requested):
Для начала, не пересоздавай PRы. Принцип PR это сравнение 2х веток. У тебя уже был PR и тебе достаточно просто внести изменения в одну из веток и PR автоматически обновится, т.к. он показывает различие между последними коммитами в ветке. Т.е. обновила ветку, обновился PR.
Так же некоторые решения у тебя все еще неверные, их надо исправить.

DamsYaYa committed:
Добавлена параметризация в тест с выпадающим списком

DamsYaYa committed:
Добавлена аннотация к закрытию браузера

DamsYaYa committed:
Добавлена аннотация к закрытию браузера

DamsYaYa committed:
Переписан тест заказа самоката, что сделано:
 1. добавлена параметризация
 2. аннотация к методу закрытия браузера
 3. браузер вынесен в качестве "настройки" в Before

pavel-ulan commented on src/main/java/ru/yandex/praktikum/ImportantQuestions.java:
```diff
@@ -1,63 +1,28 @@
-    private void checkEnableAccordionPanel(By accordionPanel){
-        Assert.assertTrue(driver.findElement(accordionPanel).isDisplayed());
+    // Получить текст ответа на вопрос
+    public String GetTexFromDropdownList(int id){
```
**можно лучше**
почему метод именован через заглавную букву? не забывай, через заглавную - классы, методы - camelCase начинающийся с прописной.

pavel-ulan commented on src/main/java/ru/yandex/praktikum/ImportantQuestions.java:
```diff
@@ -1,63 +1,28 @@
-        new WebDriverWait(driver, 3)
-                .until(ExpectedConditions.visibilityOfElementLocated(accordionPanel));
+    // Открыть стр сайта
+    public void openAWebsiteToOrderAScooter() {
```
**можно лучше**
Артиклями в именах переменных или методах пренебрегают. Вообще тема именования переменных довольна обширна и холиварна. Советую почитать что-нить по этому поводу (разной литературы и статей много), что бы лучше понимать почему и зачем существует так много негласных правил с именами.

pavel-ulan commented on src/main/java/ru/yandex/praktikum/NewOrder.java:
```diff
@@ -11,149 +10,101 @@
+    private static final By orderButton = By.xpath(".//div[@class='Order_Buttons__1xGrp']/button[text() = 'Заказать']");
+    private static final By orderConfirmation = By.xpath(".//div[@class='Order_Buttons__1xGrp']/button[text() = 'Да']");
+    private static final By successfulOrder = By.xpath(".//div[@class='Order_ModalHeader__3FDaJ' and text() = 'Заказ оформлен']");
+    private static final By finalOrderButton = By.xpath(".//button[text() = 'Заказать']");
```
**можно лучше** (хотя по хорошему надо бы исправить)
почему у тебя половина локаторов просто privat, а половина статичные константы (и раз уж ты решила сделать их константами, то почему их имена не UPPER_CASE как положено для констант?)

pavel-ulan replied:
На самом деле и тот и другой способ имеет право на жизнь (правда это всегда чем-то должно быть обосновано), но это должен быть единообразный подход. Иначе это уже что-то непонятное.

pavel-ulan commented on src/test/java/NewOrderTest.java:
```diff
@@ -12,51 +14,88 @@
-        WebDriverManager.firefoxdriver().setup();
-
-        WebDriver driver = new FirefoxDriver();
+    public void makingAnOrder(){
```
**можно лучше**
все тесты должны иметь в названии или test или check (соответствовать регексу '[A-Z][A-Za-z\d]*Test(s|Case)?|Test[A-Z][A-Za-z\d]*|IT(.*)|(.*)IT(Case)?' если быть точным). Это надо по причине того, что многие библиотеки ориентируются по названиям методов. просто в какой-то момент это может вызвать непредвиденные ошибки.

pavel-ulan reviewed (approved):
Привет.
Самое первое что бросилось в глаза - я не знаю как у тебя попали в реп файлы с ssh ключами, но это максимально секьюрная информация (максимальнее наверное нету). Т.е. сейчас - ок, ты просто открыла свой репозиторий для любого, с набором возможностей регулируемым твоим ключом. Однако в ситуации если бы это случилось с реальным проектом можно было бы вполне оказаться уволенным. Всегда следи за тем, что у тебя в коммитах, сообщениях и т.п. все пароли, ключи и прочая инфа такого плана должна быть или в специальных приложениях (vault) или только локально (если по правилам проекта так будет заведено).
Теперь к самой работе -то, что исправила - молодец 👍 
Но если откровенно - работа очень неаккуратная. Есть 100500 причин ее завернуть (как формальных, так и на основе используемых подходов). Пока вы учитесь, еще можно что-то пропустить, но и с Вашей стороны хотелось бы видеть чуть больше усердия. Даже в банальных вещах типо проверки перед отправкой на лишние импорты, неверных названий, лишних локаторов которые не используются и т.п.
С каждым заданием требования будут все строже, постарайся быть внимательнее. С такими замечаниями даже следующее задание уже не примут.
